stream_alert_cli/runner.py

```diff
@@ -141,24 +144,62 @@ def athena_handler(options):
                 list: 'array<string>'
             }
 
-            for key_name, key_type in sanitized_schema.iteritems():
-                # Transform the {} or [] into hashable types
-                if key_type == {}:
-                    key_type = dict
-                elif key_type == []:
-                    key_type = list
-
-                athena_schema[key_name] = schema_type_mapping[key_type]
-
-            schema_statement = ''.join(['{0} {1},'.format(key_name, key_type)
-                                        for key_name, key_type
-                                        in athena_schema.iteritems()])[:-1]
-            query = ('CREATE EXTERNAL TABLE {table_name} ({schema})'
-                     'PARTITIONED BY (dt string)'
-                     'ROW FORMAT SERDE \'org.openx.data.jsonserde.JsonSerDe\''
+            def add_to_athena_schema(schema, root_key=''):
+                """Helper function to add sanitized schemas to the Athena table schema"""
+                # Setup the root_key dict
+                special_key = None
+                if root_key and not athena_schema.get('root_key'):
+                    athena_schema[root_key] = {}
+
+                for key_name, key_type in schema.iteritems():
+                    # Transform the {} or [] into hashable types
+                    if key_type == {}:
```

Comment:
Should `special_key` be reset to `None` here? Or just instantiate within this loop instead of outside of it?